search-sentiments-map/routes/search.js

```diff
@@ -25,23 +25,149 @@ const datastore = new Datastore();
+// TODO(carmenbenitez): Set this to be 12 hours when we want data to be
+// updated every 12 hours.
+const CURRENT_SEARCH_RESULT_THRESHOLD_24_HOURS_MS = 24 * 60 * 60000;
+const PAUSE_TO_PREVENT_REACHING_QUOTA_1_MIN_MS = 60000;
 
 /** 
- * Renders a JSON array of the top search results for all countries with API data
- * obtained every 12 hours for the specified topic.
+ * Renders a JSON array of the top search results for all countries with API
+ * data obtained every 12 hours for the specified topic.
  */
 router.get('/:topic', (req, res) => {
   let topic = req.params.topic;
   retrieveSearchResultFromDatastore(topic).then(customSearchTopicJsonArray => {
-    // TODO(ntarn): Remove console.log statements when finished debugging.
-    console.log('ntarn debug: In the router.get method'); 
-    console.log(customSearchTopicJsonArray);
     res.setHeader('Content-Type', 'application/json');
     res.send(customSearchTopicJsonArray);
   });
 });
 
+/** 
+ * Renders a JSON array of the top search results for requested countries with
+ * API data from within the last
+ * `CURRENT_SEARCH_RESULT_THRESHOLD_24_HOURS_MS` for the specified topic.
+ */
+router.get('/:topic/:countries', (req, res) => {
+  let topic = req.params.topic;
+  let countries = JSON.parse(req.params.countries);
+  retrieveUserSearchResultFromDatastore(topic, countries).then(userSearchTopicJsonArray => {
+    res.setHeader('Content-Type', 'application/json');
+    res.send(userSearchTopicJsonArray);
+  });
+});
+
+/** 
+ * Retrieves search result data for specified countries for a given topic.
+ *   - Obtains and formats search results from Custom Search API if the data
+ *         is not currently stored.
+ *   - Saves any new search results to the datastore.
+ *   - Retrieves any relevant existing search results from datastore.
+ *   - Returns all requested search results on the topic from the countries.
+ * @param {string} topic Search topic to get data for.
+ * @param {Array} countries Countries to get search topic data for.
+ * @return {Object} JSON-formatted array of search results from requested
+ *     countries.
+ */
+async function retrieveUserSearchResultFromDatastore(topic, countries) {
+  // Request latest entity with a topic matching the given topic.
+  const query = datastore.createQuery('CustomSearchTopic').order('timestamp', {
+    descending: true,
+  }).filter('lowercaseTopic', topic.toLowerCase()).limit(1);
+  const [customSearchTopic] = await datastore.runQuery(query);
+
+  let countriesDataToReturn = [];
+  let timestamp;
+  if (customSearchTopic.length !== 0 &&
+      Date.now() - customSearchTopic[0].timestamp <
+      CURRENT_SEARCH_RESULT_THRESHOLD_24_HOURS_MS) {
+        timestamp = customSearchTopic[0].timestamp;
+        let countriesToAddDataFor = [];
+        let countriesData = customSearchTopic[0].countries;
+
+        // Determine whether a country has existing data or data needs to be
```

Comment:
Correct me if I'm wrong, but from what I understand from the styling guide,  if there's an if statement condition that needs to line wrap, I thought for the other statements in the actual if statement, you indent two only from the beginning of the if statement itself, not from the line wrapped conditions, bc if we didn't line wrap, that's how it would be?
https://google.github.io/styleguide/jsguide.html#formatting-line-wrapping

Reply:
I had that question as well

Reply:
I am also confused by this. Not sure I have heard different things. Maybe @lawrencezjh  @hayleejane3 could clear this up.